google-cloud-firestore/lib/google/cloud/firestore/query.rb

```diff
@@ -0,0 +1,652 @@
+        # provided fields.
+        #
+        # @param [FieldPath, String, Symbol] fields A field path to
+        #   filter results with and return only the specified fields. One or
+        #   more field paths can be specified.
+        #
+        #   If a {FieldPath} object is not provided then the field will be
+        #   treated as a dotted string, meaning the string represents individual
+        #   fields joined by ".". Fields containing `~`, `*`, `/`, `[`, `]`, and
+        #   `.` cannot be in a dotted string, and should provided using a
+        #   {FieldPath} object instead.
+        #
+        # @return [Query] New query with `select` called on it.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a collection reference
+        #   cities_col = firestore.col "cities"
+        #
+        #   # Create a query
+        #   query = cities_col.select(:population)
+        #
+        #   query.get do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def select *fields
+          new_query = @query.dup
+          new_query ||= StructuredQuery.new
+
+          field_refs = fields.flatten.compact.map do |field|
+            field = FieldPath.parse field unless field.is_a? FieldPath
+            StructuredQuery::FieldReference.new \
+              field_path: field.formatted_string
+          end
+
+          new_query.select ||= StructuredQuery::Projection.new
+          field_refs.each do |field_ref|
+            new_query.select.fields << field_ref
+          end
+
+          Query.start new_query, parent_path, client
+        end
+
+        ##
+        # @private This is marked private can't be removed.
+        #
+        # Selects documents from all collections, immediate children and nested,
+        # of where the query was created from.
+        #
+        # @return [Query] New query with `all_descendants` called on it.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a collection reference
+        #   cities_col = firestore.col "cities"
+        #
+        #   # Create a query
+        #   query = cities_col.all_descendants
+        #
+        #   query.get do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def all_descendants
+          new_query = @query.dup
+          new_query ||= StructuredQuery.new
+
+          if new_query.from.empty?
+            fail "missing collection_id to specify descendants."
+          end
+
+          new_query.from.last.all_descendants = true
+
+          Query.start new_query, parent_path, client
+        end
+
+        ##
+        # @private This is marked private can't be removed.
```

Comment:
Nit: Missing word - This is marked private AND/BUT can't be removed.
Here and above.

Reply:
Fixed here and above.